gui-tk/wiring.tcl

```diff
@@ -809,6 +810,46 @@ proc contextMenu {x y X Y} {
     tk_popup .wiring.context $X $Y
 }
 
+proc setupPickDimMenu {} {
+    global dimLabelPicked
+    if {![winfo exists .wiring.context.pick]} {
+        toplevel .wiring.context.pick
+        wm title .wiring.context.pick "Pick two dimensions"
+        frame .wiring.context.pick.select
+        scrollbar .wiring.context.pick.select.vscroll -orient vertical -command {
+            .wiring.context.pick.select.lb yview}
+        listbox .wiring.context.pick.select.lb -listvariable dimLabelPicked \
+            -selectmode extended -selectforeground blue \
+            -width 35 \
+            -yscrollcommand {.wiring.context.pick.select.vscroll set} 
+        pack .wiring.context.pick.select.lb -fill both  -expand y -side left
+        pack .wiring.context.pick.select.vscroll -fill y -expand y -side left
+        pack .wiring.context.pick.select
+        buttonBar .wiring.context.pick {
+            set pick {}
+            foreach i [.wiring.context.pick.select.lb curselection] {
+                lappend pick [lindex $dimLabelPicked $i]
+            }
+            minsky.canvas.item.setDimLabelsPicked [lindex $pick 0] [lindex $pick 1]
```

Comment:
You need to deal with the possibility that the user has selected less than two dimensions

Reply:
I've dealt with this, but don't know if a switch statement would be more appropriate? (Not sure if I need only two or three cases?)

Reply:
I was concerned that lindex might throw an error if the list had fewer than 2 elements. Actually, the documentation of lindex is silent on what happens here. Currently the behaviour is that an empty string will be passed to minsky.canvas.item.setDimLabelsPicked, which knowing how its used, is not disasterous. Maybe leave it for now, hopefully we'll be off TCL before lindex's error behaviour changes...

Reply:
Okay, done.